src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,140 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
```

Comment:
It's not advisable to generate such a large number of states when we only have one instance (movie). 
It'd be much better to utilize a single state with an object inside.

1. create a variable outside the Component
```
const initialMovieState = {
  title: '',
  description: '',
  imgUrl: '',
  imdbUrl: '',
  imdbId: ''
}
```

2. use one state for movie fields
3. pass `initialMovieState` object when you invoke reset `function`
4. whenever you need to change value of any field you should pass two arguments (key, value) and then just change a value of a movie object by key. In this case you shouldn't have different handlers for your fields

Example:
```
const handleInputChange = (key, value) = {
  setMovie(prevFields => {
    return {
      ...prevFields,
      [key]: value
    }
  })
}
```


```suggestion
  const [movie, setMovie] = useState(initialMovieState);
```

Reply:
https://github.com/mate-academy/react_movies-list-add-form/pull/1788/files#r1318897323
didn't fix it